pkg/operator/k8sutil/node.go

```diff
@@ -188,6 +188,40 @@ func NodeMeetsAffinityTerms(node v1.Node, affinity *v1.NodeAffinity) (bool, erro
 	return false, nil
 }
 
+// nodeSelectorRequirementsAsSelector converts the []NodeSelectorRequirement api type into a struct that implements
```

Comment:
This is copied from K8s source code since they didn't expose it anymore? We just need to add a comment here to give credit to the source, and mention why it's copied.

Reply:
right. done